components/builder-jobsrv/src/data_store.rs

```diff
@@ -284,6 +305,25 @@ impl DataStore {
         Ok(jobs)
     }
 
+    /// Reset any Dispatched jobs back to Pending state
+    /// This is used for recovery scenario
+    ///
+    /// # Errors
+    /// * If a connection cannot be gotten from the pool
+    /// * If the dispatched jobs cannot be selected from the database
+    /// * If the row returned cannot be translated into a Job
+    pub fn reset_jobs(&self) -> Result<Vec<jobsrv::Job>> {
+        let mut jobs = Vec::new();
+        let conn = self.pool.get_shard(0)?;
+        let rows = &conn.query("SELECT * FROM reset_jobs_v1()", &[])
+                        .map_err(Error::JobReset)?;
+        for row in rows {
+            let job = self.row_to_job(&row)?;
+            jobs.push(job);
+        }
+        Ok(jobs)
```

Comment:
I think it is used multiple times where you see the `?` operator, but I might be misunderstanding the question. 😅  (as usual)